### ISSUE TITLE
Implement "tab groups" feature on iOS

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#42420]
### Description

Tab groups is a popular feature that many users would like to see in the iOS build of Brave. Namely, users coming from Safari to Brave after Apple updated Safari to include tab groups as well:
https://www.macrumors.com/2021/06/11/ios-15-closer-look-at-safari/

Unless there is a significant trade off or restriction put in place by iOS preventing this, I think implementing this feature would help more iOS users make the transition to Brave. 

### Steps to reproduce

N/A

### Actual result

N/A

### Expected result

N/A

### Reproduces how often

Easily reproduced

### Brave version

v1.71.125

### Device/iOS version

All

### Affected browser versions

- [x] latest AppStore
- [x] latest TestFlight
- [ ] previous TestFlight

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] in the latest version of mobile Safari

### Miscellaneous information

Community users requesting this — will try to add additional requests I see (as I know there are many) to this thread as I find them:
https://community.brave.com/t/tab-groups-on-brave/498191/5

**Comment by Mahennz** [comment]
PLS add

**Comment by wesbrillhart** [comment]
This is the #1 feature I’d like added to iOS brave. It’s my default browser but I can’t believe this has yet to be implemented. I hope more comment and request as it’s an important feature for any modern browser! Thanks all for the hard work and excellent browser!

**Comment by StillAyo** [comment]
highly requested...

**Comment by juareza94** [comment]
Great feature preventing full use of Brave on iOS and therefore other platforms due to inability to sync between safari and Brave. Please add!

**Comment by PolQand** [comment]
This is the feature preventing me from completely switching from Safari. Much needed!

**Comment by yagni** [comment]
This would be great!

**Comment by sderungs** [comment]
It would really help transitioning fully to Brave (as a user coming from Google Chrome) on iOS as groups help keep tabs organized. 

Side note: Considering Google Chrome on iOS has tab groups I doubt there are restrictions coming from iOS.

**Comment by apolpat** [comment]
Please do it...

**Comment by AlFirous** [comment]
> Side note: Considering Google Chrome on iOS has tab groups I doubt there are restrictions coming from iOS.

Agreed, unless there's something more important to do at the moment. Tab groups are necessary, especially on mobile.

Even Chrome implements tab groups that sync properly. It would be great if Brave had this. Or just local tab groups, will be fine for most people.

**Comment by rajb16** [comment]
In my opinion this is the only feature holding brave back. Please add group tabs.

**Comment by DONTMEOWx64** [comment]
Please, implement this! Chrome has tab groups on the mobile browser, you can too!

**Comment by dvalvezon** [comment]
Having this on mobile will be beautiful! What a precious feature... !!!

**Comment by quantumcosmic** [comment]
Unfortunate that this has still not been implemented... especially if the original request was made in 2021. Hopefully this feature will be added soon but that seems unlikely at this point...

**Comment by AlFirous** [comment]
> Unfortunate that this has still not been implemented... especially if the original request was made in 2021. Hopefully this feature will be added soon but that seems unlikely at this point...

Very unlikely, considering it's labeled P5 Priority, which means it's not scheduled and at the bottom of the priority list. Unless something magical happens.

**Comment by LukasExemplar** [comment]
After security issues, this should be the top priority. Researching and using a browser most effectively, or effectively at all, hinges on organisational features of this kind, more so this specific feature than any other organisational feature that could be added. As long as it is added and in a way that won’t slow the browser app down noticeably, I’d be haply!

**Comment by reido113** [comment]
This feature would have me using Brave 100%.

**Comment by kapods22** [comment]
+100

**Comment by trudog** [comment]
Tab organization is essential. Open tabs and the ability to search them is an amazing feature, and is quickly replacing the usefulness of traditional browser bookmarks. Just like the evolution of bookmarks from a simple list to a hierarchy of folders for managing and organizing, the same feature is becoming a hard requirement for tabs. 

The more I use brave on the desktop and mobile, the more open tabs (bookmarks) I have open and I need a way to group and organize them.  I love the Brave browser but I'm already considering looking for "secure" alternatives that offer tab grouping on iOS.  

Please bump the priority of this feature request before the completion steals your loyal customer base. 

Thank you.

**Comment by hakimsjo** [comment]
This is highly requested. +1

**Comment by Westley33** [comment]
PLEASE. This would be so helpful and would definitely get me to delete Safari from my phone.

**Comment by radical-ed** [comment]
+1 from me! It is holding me from moving fully to brave from Safari :(

**Comment by ZTzen** [comment]
Much needed

**Comment by maestro-bene** [comment]
I would love to see this and completely switch as well

**Comment by dpangier** [comment]
Surprised this feature is missing having just tried bringing over all my open tabs from Safari. How much demand does there need to be?

**Comment by tomas-herman** [comment]
Hi @jamesmudgett @Brave-Matt, any chance to bump this up in priority? As this feature has been requested for many years by a huge amount of people? Since this features works both on MacOS and Android, it would be awesome to have it also for iPhone. Is there anything the community can do to support this feature?

**Comment by reido113** [comment]
Every time that I get an email alert from this thread I cross my fingers in hopes that I open the email to see "Finally, Tab Groups Has Been Implemented In Brave iOS!"

Anyone else???

#bringTabGroupsToBraveIos

**Comment by rajb16** [comment]
same lol
I get my hopes up but it's just another request

**Comment by wesbrillhart** [comment]
Same for me, I want it to be my main browser on the iPad so badly. Haven't
gotten another iOS browser to avoid all ads, including youtube vids. I
found a trick to load youtube through VPN to an obscure location that
doesn't implement ads (Algeria for me). Then I won't have ads on YT on
chrome for a while after disconnecting the VPN.

**Comment by mfilipe** [comment]
I use tab groups as a way to manage my tabs and unfortunately it isn’t available in iOS yet. Please, implement it!!!

**Comment by AgileBuckle** [comment]
Would love if this was implemented. Would be lovely if they synced but any sort of tab groups would be amazing.

**Comment by WilliamJohnDavisIII** [comment]
Sorry to get everyone's hopes up but I'm just here to say that this feature would be incredibly helpful. I use my iPhone to research various topics on the internet and with Brave I end up having 20+ random tabs that I have to sort through instead of tab groups I can neatly organize.

**Comment by gshari89** [comment]
I would love to see this feature!

**Comment by MrHHHHHKKKKKK** [comment]
Ohhh plz add!!! Plz!!! I love you brave!!!

**Comment by mlntr** [comment]
It's so annoying to browse without this feature 😩 My tab list has become so messy.

**Comment by atomic004** [comment]
I don't understand this browser, so much extra marginal useless functions, but tab groups don't sync from OS X to iOS because they don't exist in iOS? no vertical tabs either. It seems to have a great account free sync, but nothing useful syncs to iOS, crazy.

why would anyone move away from safari if this is the alternative? what is the pitch here? not more private than safari, certainly uglier, and no sync. just weird.

**Comment by jldevezas** [comment]
@atomic004 I agree, tab groups should sync on iOS. They seem to giving it no attention. Regarding vertical tabs, they do have it on desktop, at least in the Windows version. It works well.

**Comment by randomuser63** [comment]
Yes why isn’t this a thing. It’s why I can’t stick to brave for long, I would be a dedicated premium customer if this was an option but it’s not worth the money if I can’t do something this basic.

**Comment by auslegungssache** [comment]
bump. the lack of this feature is a complete showstopper

**Comment by Saoiray** [comment]
@bsclifton I recently saw someone asking about this on Reddit. Safari seems to allow users to manage tab groups but I'm not seeing the option in Brave or Chrome, at least not yet. 

Think we can get anyone from iOS team to revisit this and look at potentially introducing it here?

**Comment by rajb16** [comment]
Could one of the devs at least let us know if this is planned or being worked on? Or if it's even possible. Thx

**Comment by jldevezas** [comment]
> @bsclifton I recently saw someone asking about this on Reddit. Safari seems to allow users to manage tab groups but I'm not seeing the option in Brave or Chrome, at least not yet. 
> 
> Think we can get anyone from iOS team to revisit this and look at potentially introducing it here?

Chrome does support tab groups on iOS. It's implemented on the tab switching window, as a button on top. There has been an argument that there's some sort of technical limitation that is specific to iOS, but there's not. Otherwise only Safari would support tab group syncing. At this point, I just think that Brave either doesn't like Apple or doesn't have anyone to assign this work to.

**Comment by Saoiray** [comment]
Just had a reply from one of the other devs I messaged. Their reply is below:

> While we do inherit the models and the syncing aspect of tab groups, we don’t use Chrome’s UI so there’s already a significant step to support it. On top of that we don’t currently use Chromium for tab management (we started a Firefox fork, not Chromium), so we’re not inheriting the logic of managing those tab groups yet either. You’ll see feature flags for everything that’s in Chrome because we don’t attempt to prune it based on what we’re using but a majority of the Chrome specific ones will do nothing. The tab group feature is niche and we have other priorities so it’s been backlogged for now but as we migrate more core browser logic over to Chromium accessing these features become easier, or at least the scope of them will typically be reduced to UI implementations. Hope this helps clear things up!


So I am a bit confused on it. I didn't see tab groups on Firefox on iOS when I just tested, so if they are still using it then it might make sense. Regardless, I'm seeing that they have this on the roadmap but it's a low priority and they have a lot of other things their attention is focused on. We'll likely see it down the road but no ETA.

**Comment by rajb16** [comment]
I'm sorry but it's not "niche". In today's browser it's an essential feature and the only reason why I don't use brave.

I remember devs talking about how it was on the roadmap 3 years ago and nothing has changed since I guess.

**Comment by jldevezas** [comment]
@Saoiray That does clear things a bit. If they're undergoing a migration, I understand the pressure they're probably under, but it's like @rajb16 said, this is not niche. For me personally, all my organization depends on tab groups nowadays, and I know a lot of other people who also depend on this heavily. I'll check back next year, I guess. What's one more year? 😅

**Comment by Saoiray** [comment]
@jldevezas, @rajb16, and whoever else....don't forget that niche means that only a relatively small fraction of users adopt the feature in their regular workflow. 

Both of you replied about how it's important to you, but keep in mind Brave and others are tracking how many people use features. They also keep an eye out on how many people make requests. I'm guessing they just don't see "enough" of the ~100 million or so monthly active users requesting or using the functionality in order to justify the extra work on it for now. But as said, it will eventually come. They just have some other higher priorities that would be more beneficial to users and the browser in general. 

Anyway, don't forget I'm just another user. So take my opinion with a grain of salt. We did get another round of mentions to people at Brave and we'll see how things go. At least they confirmed it's something on their radar, just not necessarily something with a high priority and is more of a backburner for now.

**Comment by rajb16** [comment]
Only a niche group of people use brave. Only a niche group of people care about ads(yes, most people have no clue what an ad block even is or what extensions are). IMO anyone who uses brave automatically knows more about browsers and the features that browsers offer. So I disagree that brave users who want tab groups are a niche group, I'd argue we are the majority. 

Edit: To put it in perspective, the iOS browser "Quiche" is made by just one person and it features tab grouping based on the URL.

**Comment by jldevezas** [comment]
@Saoiray Ok, but let's be a bit fair here. One thing that they can't track is the amount of users that would switch to Brave if it had tab group sync. Some of them don't care enough to comment on it either. I was in the first group, up to when Chrome went to shit with their ad blockers being broken. From what I've read so far, it seems there are more people that, like me, need that feature so much that they won't do the switch. It's hard to quantify potential. It's also hard to predict where the browser market will be next year. This year, we've already seen vertical tabs getting the hype (and Brave did implement this right away), and maybe next year there will be a privacy-focused browser that does a great job with tab groups, without being bundled with crypto stuff. That said, it's fine with me either way. At least we now have some feedback on this. It's delayed because of an ongoing migration to chromium.

**Comment by atomic004** [comment]
Switching away from safari is niche, but people who switch browsers want to sync their tabs. I am looking for a browser that can sync tabs across Windows, macOS, and iOS. Tab syncing is not niche in braves target public.

Most "normal," non-niche users will use Safari or sometimes—out of habit or due to being in the Google ecosystem—Chrome. Funny enough those normal users have automatic tab syncing without even thinking about it.

Why would anybody switch to Brave? To give up iCloud integration (passwords and Hide My Email)? To lose syncing of vertical tabs? For the uglier interface? Why? There’s no reason the apple versions of brave even exist.

If it's technically impossible for Brave to make a functioning product, they should just stop developing Apple apps and focus on Windows-only users. On Windows only, I would use this or Vivaldi. But in reality, once you have even a single Apple device, why would anyone switch? What functionality does Safari lack that Brave offers? Who is this for? 

btw this is not a brave only problem, Vivaldi, Opera and even edge don't sync tabs either. 
Maybe their target market is people who use a second browser to hide things from their partner?

**Comment by BloodyRonin** [comment]
Please figure out a way to implement this. As others have said, it’s the only reason I can’t fully switch from Safari to Brave.

**Comment by namebrand** [comment]
I want to make the switch to brave as well but tab groups is holding me back!

**Comment by nicolabotta** [comment]
Without being able to manage and sync groups of tabs between iOS / iPadOS and other platforms, Brave is not really platform independent. This functionality is mandatory for using Brave in iOS / iPad OS and it is still unsupported, unfortunately.

**Comment by dev27git** [comment]
Still there's no update on this?

**Comment by mlntr** [comment]
> Still there's no update on this?

Because @jamesmudgett set this issue to P5 (the lowest priority level), so they're not actively considering/looking into it unfortunately

**Comment by sauloAH** [comment]
Keeping the dream alive! Please add this feature it, all my traffic could be concentrated in Brave after that

**Comment by longsh07** [comment]
Disappointing to see this feature isn’t on iOS as it’s something I use heavily on Android and Windows versions of Brave. Surprised its not considered a higher priority for feature parity.

**Comment by Arenovas** [comment]
I just recently switched from Android to iOS and man it sucks not having tab groups, honestly surprising it isn’t already in by now.

**Comment by 6554ypknr4-pixel** [comment]
Please please please add it as soon as you can. Especially now that Brave is the only browser to let us choose wether we want an AI overview or not, the “tab groups” feature would help so many ppl make the switch to brave and not use AI. 
This is a matter of saving our water ressources all over the world. We need you brave !

**Comment by vardecab** [comment]
> now that Brave is the only browser to let us choose wether we want an AI overview or not

[Quiche Browser](https://quiche.industries/browser/) added this a few weeks ago: https://quiche.industries/browser/whatsnew/
<img width="518" height="135" alt="Image" src="https://github.com/user-attachments/assets/7f237dc9-74f2-451a-a148-12b8e4f602fc" />

**Comment by fxreeda** [comment]
pls

**Comment by CaptainCuddlesJ** [comment]
Please add this, it is actually wild that this isn’t a feature. Unfortunately will be changing browsers until Brave adds this….
